datafusion/sqllogictest/test_files/functions.slt

```diff
@@ -815,3 +815,33 @@ SELECT products.* REPLACE (price*2 AS price, product_id+1000 AS product_id) FROM
 1002 OldBrand Product 2 59.98
 1003 OldBrand Product 3 79.98
 1004 OldBrand Product 4 99.98
+
+#overlay tests
+statement ok
+CREATE TABLE over_test(
+  str TEXT,
+  characters TEXT,
+  pos INT,
+  len INT
+) as VALUES
+  ('123', 'abc', 4, 5),
+  ('abcdefg', 'qwertyasdfg', 1, 7),
+  ('xyz', 'ijk', 1, 2),
+  ('Txxxxas', 'hom', 2, 4)
```

Comment:
Can you please also test `NULL` handling here? I think your code handles it properly, but it would be good to have this in the SQL tests. We can do this as a follow on PR if you prefer. 

Specifically
```suggestion
  ('Txxxxas', 'hom', 2, 4),
  (NULL, 'hom', 2, 4), -- expect NULL output
  ('Txxxxas', 'hom', NULL, 4), -- expect NULL output
  ('Txxxxas', 'hom', 2, NULL) -- expect NULL output
```

Reply:
Have added these cases, thanks.